crates/transaction-pool/src/validate/eth.rs

```diff
@@ -18,24 +18,32 @@ use reth_primitives::{
     ChainSpec, InvalidTransactionError, SealedBlock, EIP1559_TX_TYPE_ID, EIP2930_TX_TYPE_ID,
     EIP4844_TX_TYPE_ID, LEGACY_TX_TYPE_ID,
 };
-use reth_provider::{AccountReader, StateProviderFactory};
+use reth_provider::{AccountReader, BlockReaderIdExt, StateProviderFactory};
 use reth_tasks::TaskSpawner;
 use std::{
     marker::PhantomData,
     sync::{atomic::AtomicBool, Arc},
 };
 use tokio::sync::Mutex;
 
+#[cfg(feature = "optimism")]
+use reth_primitives::{bytes::BytesMut, BlockNumberOrTag};
```

Comment:
`BytesMut` is only used once, remove this import and use the absolute dependency path inline.

Reply:
Same thing with `BlockNumberOrTag`

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/154